### PR TITLE
JengaBase get request

### PR DESCRIPTION
**Pull request by r-coh**
## What is the Purpose?

Handling of `GET` request in JengaBase, as mentioned [in this pr](url)

## Issue(s) affected?

Required by #60 and #81

### CODE REVIEW

File: lib/ex_pesa/Jenga/jenga_base.ex [hunk 1]
```diff
@@ -97,6 +97,44 @@ defmodule ExPesa.Jenga.JengaBase do
...
+  @doc """
+  performs `GET` requests.
+  `get_request/2` accepts, parameters encoded url and headers
+  `get_request/3` accepsts, url, body as map and headers
+  """
+  @spec get_request(bitstring(), keyword()) :: {:ok, map()} | {:error, bitstring()}
+  def get_request(url, headers) when is_list(headers), do: get_request(url, nil, headers)
+
+  @spec get_request(bitstring(), map(), keyword()) :: {:ok, map()} | {:error, bitstring()}
+  def get_request(url, body, headers \\ [])
```
Comment (lenileiro) [inline lib/ex_pesa/Jenga/jenga_base.ex:109]:
```suggestion
```

File: lib/ex_pesa/Jenga/jenga_base.ex [hunk 2]
```diff
@@ -97,6 +97,44 @@ defmodule ExPesa.Jenga.JengaBase do
...
+
+  def get_request(url, body, headers) do
```
Comment (lenileiro) [inline lib/ex_pesa/Jenga/jenga_base.ex:111]:
```suggestion
  def get_request(url, body, headers \\ []) do
```

File: lib/ex_pesa/Jenga/jenga_base.ex [hunk 3]
```diff
@@ -97,6 +97,44 @@ defmodule ExPesa.Jenga.JengaBase do
...
+    # anonymous function get./1 to parse body if provided
+    get = fn client ->
+      opts = [adapter: [recv_timeout: 30_000]]
```
Comment (lenileiro) [inline lib/ex_pesa/Jenga/jenga_base.ex:114]: you wanted to set the recv_timeout on Tesla?

Comment (r-coh) [reply lib/ex_pesa/Jenga/jenga_base.ex:114]: yeah, to avoid repeating the `opts` when calling Tesla.get below.

Comment (lenileiro) [reply lib/ex_pesa/Jenga/jenga_base.ex:114]: @r-coh Just repeat it for now, we will optimise later

Comment (r-coh) [reply lib/ex_pesa/Jenga/jenga_base.ex:113]: @lenileiro alright.

Comment (manuelgeek) [reply lib/ex_pesa/Jenga/jenga_base.ex:113]: there's an option to pass it with the configs but the implantation i did it was somehow not working, then about making headers not optional, i think we can just make it optional , we;re not sure of the nature og all get requests we'll meet

**Comment by manuelgeek** [comment]
@r-coh  kindly squash your commits to 1, so that i can easily review and approve, kindly

**Comment by r-coh** [comment]
@manuelgeek all good!

**Comment by manuelgeek** [comment]
@r-coh one. more thing rebase, haha. then we're all good

**Comment by r-coh** [comment]
@lenileiro Hahaha, adi refactor umeitag `chore`
..sorry about that.

**Review by lenileiro — APPROVED** [review]
🔥🔥

**Pull request merged**